openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -180,3 +206,60 @@ def _email_url_parser(url_name, extra_param=None):
     dashboard_url_path = reverse(url_name) + extra_param if extra_param else reverse(url_name)
     dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
     return urlparse.urlunparse(dashboard_link_parts)
+
+
+def get_credit_providers_by_course(user, course_key):
+    """Get the course information from ecommerce and parse the data to get providers.
+
+    Arguments:
+        user (User): The user object.
+        course_key (CourseKey): The identifier for the course.
+
+    Returns:
+        list, containing course providers.
+    """
+    credit_config = CreditConfig.current()
+
+    # Bypass caching for staff users, who may be adding providers and want
+    # to see them immediately.
+
+    providers_data = None
+    cache_key = None
+    if credit_config.enabled:
+        cache_key = '{key_prefix}.{course_key}'.format(
+            key_prefix=credit_config.CACHE_KEY, course_key=unicode(course_key)
+        )
+        providers_data = cache.get(cache_key)
+
+    if providers_data is not None:
+        return providers_data
+
+    providers_data = []
+    try:
+        response = ecommerce_api_client(user).courses(unicode(course_key) + '/?include_products=1').get()
```

Comment:
@rlucioni this method uses [get_id_token](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/token_utils.py#L12). And it uses preferred-username in payload which is different from ecommerece. I have tried to pass username in payload aswell but it fail. One thing I tried also I passed the ECOMMERCE_SERVICE_WORKER_USERNAME user object in [ecommerce_api_client](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/commerce/utils.py#L29) and it works. But for this we need to define its userprofile or modify or create new one.
@clintonb would you please suggest how to access course api in ecommerece. This api need admin permission.